### PR TITLE
#62 Client now restarts after sync failures

### PR DESCRIPTION
**Pull request by glothriel**

### CODE REVIEW

File: pkg/hello/syncing.go [hunk 1]
```diff
@@ -173,7 +182,7 @@ func NewHTTPSyncingClient(
 	if !ok {
 		return nil, errors.New("sync_server_address not found in pairing response metadata")
 	}
-	transport := NewHTTPClientSyncingTransport(syncServerAddress)
+	transport := NewHTTPClientSyncingTransport(syncServerAddress, 3*time.Second)
```
Comment (jzyinq) [inline pkg/hello/syncing.go:185]: Ideally this should be configurable through helm or extracted as const, I think it's ok for now.

File: pkg/hello/syncing.go [hunk 2]
```diff
@@ -150,12 +158,13 @@ func NewSyncingClient(
...
-		interval:     interval,
-		apps:         apps,
-		transport:    transport,
+		myName:               myName,
+		stateChangeGenerator: nginxAdapter,
+		encoder:              encoder,
+		interval:             interval,
+		apps:                 apps,
+		transport:            transport,
+		failureThreshold:     3,
```
Comment (jzyinq) [inline pkg/hello/syncing.go:167]: A candidate for const / configurable value through helm.

**Review by jzyinq — APPROVED** [review]
I'm missing some information in logs after `log.Fatal` that command is being restarted. 

Btw how is it that command fails yet pod is not being restarted but process seems to handle it internally?

EDIT:

OK that's because of [dev-entrypoint](https://github.com/glothriel/wormhole/blob/34-restart-k8s-watcher-every-5-minutes/docker/go/dev-entrypoint.sh#L22) - so in real life scenario this will restart the pod right?

File: pkg/hello/syncing.go (hunk 1 above)
Comment (glothriel) [reply pkg/hello/syncing.go:185]: Agreed, leaving for now

File: pkg/hello/syncing.go (hunk 2 above)
Comment (glothriel) [reply pkg/hello/syncing.go:167]: Agreed, leaving for now

**Comment by glothriel** [comment]
As to those parameters. I think I'll create an issue to dehardcode those random parameters. I try to put them in the constructions si i have them already plucked out to some extent.

**Pull request merged**